src/components/Calendar.tsx

```diff
@@ -205,38 +269,39 @@ export const Calendar = ({
     }, [refresh])
   );
 
-  const fetchMoreData = () =>
-    fetchMoreDateRange({
-      query: getFetchMoreQuery(query),
-      variables: {
-        ...queryVariablesWithDateRange,
-        offset: queryVariablesWithDateRange.limit
-      },
-      updateQuery: (prevResult, { fetchMoreResult }) => {
-        if (!fetchMoreResult?.[query]?.length) return prevResult;
+  const fetchMoreData = useCallback(() => {
+    if (hasNextPageSubList) {
+      return fetchNextPageSubList();
+    }
 
-        const uniqueData = _uniqBy([...prevResult[query], ...fetchMoreResult[query]], 'id');
+    return { data: { [query]: [] } };
+  }, [dataSubList, fetchNextPageSubList, hasNextPageSubList]);
 
-        return {
-          ...prevResult,
-          [query]: uniqueData
-        };
-      }
-    });
+  const disableArrowLeft =
+    (today || moment().startOf('month').format('YYYY-MM-DD')) ===
+    queryVariablesWithDateRange.dateRange[0];
 
   return (
     <>
       <RNCalendar
         dayComponent={DayComponent}
-        displayLoadingIndicator={loading || refreshing}
+        disableArrowLeft={disableArrowLeft}
+        displayLoadingIndicator={loading || isRefetching || isRefetchingSubList}
         firstDay={1}
         markedDates={markedDates}
         markingType="multi-dot"
+        minDate={today}
         onDayPress={onDayPress}
-        renderArrow={renderArrow}
+        onMonthChange={onMonthChange}
+        renderArrow={(direction: Direction) =>
```

Comment:
I'm sorry but I can tell you that it's still not working correctly. There is no problem with the list view. But the calendar view is still problematic. I will try to explain with screenshots.

I have added a new event. This event runs from 01.07.2024 until 31.08.2024 and repeats only on Tuesdays.  If I go to the Event screen, we can see this in the list. For example, there is one of this event on 30.07.2024. But when I go to the calendar view, I don't see a dot for the event on 30.07.2024 and I can't click on it. Then I click on another date (e.g. 13.08.2024). Then a dot starts to appear under the date 30.07.2024. Then I click on the date 30.07.2024 and I see 2 events, but in the list view only 1 event appears.

As far as I understand, for a recurring event to appear in the calendar, there must be more than one event on that day or the system automatically adds any event to the same day (I am not sure about this).

|||||
|--|--|--|--|
![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 30](https://github.com/user-attachments/assets/566b331b-a1ad-4c18-94a7-47d3f27c0cd2)|![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 39](https://github.com/user-attachments/assets/ec033d9b-d339-44c3-8e8b-e5a59141eee6)|![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 20 56](https://github.com/user-attachments/assets/992c6fed-e37f-4561-8427-5c1bdffcc4a1)|![Simulator Screenshot - iPhone 15 Pro Max - 2024-07-26 at 10 21 03](https://github.com/user-attachments/assets/79e583ae-0a17-49c4-b843-45be5010704c)

Reply:
i can't reproduce that. as we chatted about this a bit, do you think this is still a problem?

Reply:
The cause of the problem is just a little confusion. We discussed and resolved this issue on Slack.